src/Meta/MetaFactory.cpp

```diff
@@ -193,15 +193,14 @@ void MetaFactory::createFromFunction(const clang::FunctionDecl& function, Functi
     bool returnsNotRetained = Utils::getAttributes<clang::NSReturnsNotRetainedAttr>(function).size() > 0 || Utils::getAttributes<clang::CFReturnsNotRetainedAttr>(function).size() > 0;
 
     // Clang doesn't handle The Create Rule automatically like for methods, so we have to do it manually
-    if (!(returnsRetained || returnsNotRetained)) {
-        Type* returnType = functionMeta.signature[0];
-        if (returnType->is(TypeInterface) || returnType->is(TypeBridgedInterface) || returnType->is(TypeId)) {
-            std::string functionName = function.getNameAsString();
+    if (!(returnsRetained || returnsNotRetained) && functionMeta.signature[0]->is(TypeBridgedInterface)) {
```

Comment:
Why did we decide not to memory manage `TypeInterface` and `TypeId`? It seems that now e.g. `dispatch_queue_t CFReadStreamCopyDispatchQueue(CFReadStreamRef stream)` isn't memory managed but [ARC is able to do so](http://stackoverflow.com/questions/8618632/does-arc-support-dispatch-queues).